Source/WebKit/UIProcess/AuxiliaryProcessProxy.cpp

```diff
@@ -311,6 +318,15 @@ void AuxiliaryProcessProxy::didFinishLaunching(ProcessLauncher*, IPC::Connection
     }
 }
 
+void AuxiliaryProcessProxy::outgoingMessageQueueIsGrowingLarge()
+{
+#if USE(RUNNINGBOARD)
+    // If we keep trying to send IPC to a suspended process, the outgoing message queue may grow large and result
+    // in increased memory usage. To avoid this, we wake up the process for a bit so we can drain the messages.
+    m_timedActivityForIPC = throttler().backgroundActivity("IPC sending due to large outgoing queue"_s);
```

Comment:
This gets called back on the IPC send queue, does it need to dispatch to the main thread?

Reply:
Oh wait, never mind, seems like this gets called on the thread that tried to send the IPC. Which I guess in most cases is the main thread. Are there cases where we try to send IPCs off the main thread in UIProcess?

Reply:
This doesn't get called on the IPC send queue, it gets called on the same thread as `send()`. That said, it is true that `send()` can be called off the main thread. I'll refine.